src/M.java

```diff
@@ -0,0 +1,38 @@
+class B {
+    private String shape;
+    private int a, b;
+    public B(String s, int x, int y) {
+        shape = s;
+        a = x;
+        b = y;
+    }
+
+    public String toString() { 
+        return shape + ":area=" + area(); 
+    }
+
+    protected int area() { 
+        return a*b; 
+    }
+}
+
+class D extends B {
+
+    public D(int x) {
+        super("Square", x, x);
+    }
+}
+
+public class M {
+    public static void main(String[] args) {
+        B[] b = {
```

Comment:
Using a one letter variable names is usually a bad practice (apart from "i" as a shortcut for iteration in a for loop).

Maybe rename b to "arr"